beastling/distributions.py

```diff
@@ -1,9 +1,11 @@
 import sys
 import math
-import collections
+
+import attr
```

Comment:
python 3.5/3.6 compat, mainly. But also the better integration of parsing and validating - via `converter` and `validator` callables.

Reply:
Good to know there are currently other advantages for attrs beyond compatibility. For other projects, where compatibility is not as pressing, I have started using dataclass in favour of attrs because I thought I should prefer builtins, but it's good to know its limitations.

Reply:
It's a mixed bag - in particular considering the weird compat problems with python/attrs/pytest on travis. But for most of my uses - wrapping human-user-supplied data with a bit of functionality - conversion and validation always were an issue. Here's a nice overview: https://www.revsys.com/tidbits/dataclasses-and-attrs-when-and-why/